src/refinement/triquadrisection.jl

```diff
@@ -0,0 +1,65 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    TriQuadrisection()
+
+Refinement of a mesh by preliminarly triangulating it if needed and
+then subdividing each triangle into four triangles.
+
+## References
+
+* Charles Loop. 1987. [Smooth subdivision surfaces based on 
+  triangles](https://charlesloop.com/thesis.pdf). 
+  Master's thesis, University of Utah.
+"""
+struct TriQuadrisection <: RefinementMethod end
+
+function refine(mesh, ::TriQuadrisection)
+  if !(eltype(mesh) <: Triangle)
+    tmesh = simplexify(mesh)
+  else
+    tmesh = mesh
+  end
+
+  # retrieve vertices
+  points = vertices(tmesh)
+  npoints = length(points)
+
+  # convert topology to half-edge structure
+  t = convert(HalfEdgeTopology, topology(tmesh))
+  ntriangles = nelements(t)
+
+  # add middle points of edges
+  middles = Dict{Tuple{Int,Int},Int}()
+  ∂₁₀ = Boundary{1,0}(t)
+  for eind in 1:nfacets(t)
+    e1, e2 = ∂₁₀(eind)
+    p1, p2 = points[e1], points[e2]
+    midpoint = Point((coordinates(p1) + coordinates(p2))/2)
```

Comment:
```suggestion
    midpoint = center(Segment(p1, p2))
```

Reply:
`center` is not implemented for `Segment`. Can I implement it in this PR? Or I open a new one? Hmm, yes, I think a new one is better.